make/configure.mk

```diff
@@ -51,10 +52,19 @@ MERGED_DEPENDENCIES        := \
   $(TEST_DEPENDENCIES) \
   $(PLUGIN_DEPENDENCIES) \
   $(PLUGIN_SHARED)
+
 UNIQ_MERGED_DEPENDENCIES   := $(call uniq, $(MERGED_DEPENDENCIES))
 DEPENDENCIES                = $(UNIQ_MERGED_DEPENDENCIES)
 FEATURES                   := $(sort $(call subtraction,$(SUB_FEATURES),$(DEFAULT_FEATURES) $(ADD_FEATURES)))
 
+ifeq ($(UI),1)
+  MERGED_DEPENDENCIES += $(UI_DEPENDENCIES)
+else ifeq ($(TEST),1)
+  $(error "Can TEST=1 can not be set when UI=0")
+else ifeq ($(call fcheck,clap jack vst2 vst3,$(FEATURES),1),1)
+  $(error "Can not set UI=0 when FEATURES contains clap, jack, vst2 or vst3")
```

Comment:
I think it's only `gst` and `lv2` that can be built without UI support. I have no way of testing anything other than `lv2`